src/fucntions.js

```diff
@@ -1,31 +1,56 @@
 class Tasks {
-  constructor(id, description, completed) {
-    this.id = id;
+  constructor(index, description, completed) {
+    this.index = index;
     this.description = description;
     this.completed = completed;
-    this.taskList = [
-      {
-        description: 'Do Hackerrank Challenge',
-        completed: false,
-        index: 3,
-      },
-      {
-        description: 'Play Fortnite',
-        completed: true,
-        index: 2,
-      },
-      {
-        description: 'Read my bible',
-        completed: true,
-        index: 1,
-      },
-    ];
+    this.taskList = [];
+  }
+
+  getStorage() {
+    this.taskList = JSON.parse(localStorage.getItem('Tasks')) || [];
+    return this.taskList;
   }
 
   setStorage() {
-    if (localStorage.getItem('Tasks') == null) {
-      localStorage.setItem('Tasks', JSON.stringify(this.taskList));
+    localStorage.setItem('Tasks', JSON.stringify(this.taskList));
+  }
+
+  addItem() {
+    const task = {
+      index: this.index,
+      description: this.description,
+      completed: this.completed,
+    };
+
+    this.taskList = this.getStorage();
+    this.taskList.push(task);
+    this.setStorage();
+  }
+
+  updateItem(id, text) {
+    id = Number(id);
+    this.taskList = this.getStorage();
+    for (let i = 0; i < this.taskList.length; i += 1) {
+      if (this.taskList[i].index === id) {
+        text.addEventListener('change', (e) => {
+          this.taskList[i].description = e.target.value;
+          this.setStorage();
+        });
+      }
     }
   }
+
+  deleteAllCheckedItems(tasks) {
+    this.taskList = tasks;
+    this.taskList = this.getStorage();
+    this.taskList = this.taskList.filter((task) => task.completed === false);
+    this.setStorage();
+  }
+
+  deleteSingleItem(id) {
+    this.taskList = this.getStorage();
+    this.taskList = this.taskList.filter((task) => task.index !== id);
+    this.setStorage();
+  }
```

Comment:
- [ ] Good job on creating the functionality 👍 however according to the requirements deleting a task should update all remaining items' indexes, so they represent the current list order and are unique. right now when the user deletes an item from the middle of the list the index of the other items don't get updated.